pandas/tests/generic/test_generic.py

```diff
@@ -948,3 +947,21 @@ def test_axis_classmethods(self, box):
             assert obj._get_axis_number(v) == box._get_axis_number(v)
             assert obj._get_axis_name(v) == box._get_axis_name(v)
             assert obj._get_block_manager_axis(v) == box._get_block_manager_axis(v)
+
+    # Tests that the ignore keyword causes the intended functionality
+    def test_astype_ignores(self):
```

Comment:
Nitpick: you might want to consider pytest.mark.parametrize
https://docs.pytest.org/en/latest/parametrize.html#pytest-mark-parametrize

You can use `from contextlib import ExitStack as does_not_raise` as a context manager for the final case as per http://doc.pytest.org/en/latest/example/parametrize.html#parametrizing-conditional-raising

Reply:
I would love to learn more about pytest.mark.parametrize in the future but I'm not super confident implementing it here without trying it out in other smaller/less important projects first.  If you think it is absolutely essential then I will push another commit once I figure it out.